### PR TITLE
Enforce retention of field_arrays for all filters

### LINKED ISSUE #930: extract_cells filter does not keep text fields.
The picking method with `visible_pick_call_back ` does not keep text fields in arrays anymore.

The `visible_pick_call_back ` depends on `extract_cells`, line 162 in picking.py:

`picked.append(smesh.extract_cells(cids_to_get))`

after this line, all my text arrays are stripped. I noticed that I had to remove .decode() in my code because text arrays are not binary anymore in pyvista, so it may have to do with that change. I use text arrays to recover the name and category of the data picked (top app categories, not pyvista data types).

### PR DESCRIPTION
**Pull request by math-artist**
This adds a condition that will detect when the data/output has empty field_arrays but its original input has field_arrays, and update the field_arrays when this happens.

Resolves #930

### CODE REVIEW

File: pyvista/core/filters.py [hunk 1]
```diff
@@ -54,6 +54,8 @@ def _get_output(algorithm, iport=0, iconnection=0, oport=0, active_scalars=None,
     data = wrap(algorithm.GetOutputDataObject(oport))
     if not isinstance(data, pyvista.MultiBlock):
         data.copy_meta_from(ido)
+        if not data.field_arrays and ido.field_arrays:
+            data.field_arrays.update(ido.field_arrays)
```
Comment (akaszynski) [inline pyvista/core/filters.py:58]: Looks like the tests are missing coverage for L58.  Can we introduce a test for this?

Comment (banesullivan) [reply pyvista/core/filters.py:58]: I just added a test on the `math-artist-fix_field` I thought that would push to this PR.... it has in the past 😕

Comment (banesullivan) [reply pyvista/core/filters.py:58]: Hol up. I'm reteaching myself how to do this... I will have the cahnge on this PR momentarily

Comment (banesullivan) [reply pyvista/core/filters.py:58]: Got it! For some reason that was more complicated than I remember it being

**Review by akaszynski — APPROVED** [review]
LGTM.   Thanks for adding the test as well @banesullivan

**Pull request merged**